instrumentation/internal/internal-lambda-java9/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/internal/lambda/Java9LambdaTransformer.java

```diff
@@ -19,6 +20,10 @@ private Java9LambdaTransformer() {}
       String slashClassName,
       Class<?> targetClass)
       throws IllegalClassFormatException {
+    // Skip transforming lambdas of injected helper classes.
+    if (InjectedHelperClassDetector.isHelperClass(targetClass)) {
+      return classBytes;
+    }
```

Comment:
Don't we need to check that in the `LambdaTransformer` too? (or instead)

Reply:
Yes, it should be moved to `LambdaTransformer`